src/plugins/intel_gpu/tests/unit/test_cases/convolution_gpu_test.cpp

```diff
@@ -10454,6 +10454,121 @@ TEST(convolution_gpu_onednn, grouped_runtime_weights) {
     ASSERT_EQ(output_layout.get_shape(), ov::Shape({1, 256, 25, 25}));
 }
 
+TEST(convolution_gpu_onednn, dyn_conv4d_reshape6d_pattern) {
+    tests::random_generator rg(GET_SUITE_NAME);
+    auto& engine = get_test_engine();
+    if (!engine.get_device_info().supports_immad)
+        return;
+
+    if (!engine.get_device_info().supports_fp16)
+    {
+        std::cout << "[ SKIPPED ] The test is skipped (cl_khr_fp16 is not supported)." << std::endl;
+        ASSERT_EQ(1, 1);
+        return;
+    }
+
+    const int input_x = 8, input_y = 8, input_f = 32, output_f = 32, filter_x = 3, filter_y = 3, groups = 1, batch_num = 1;
+    const uint64_t stride = 1;
+    auto input_data_format = format::bfyx;
+
+    auto input_size = tensor(batch_num, input_f, input_x, input_y);
+    auto input_data = rg.generate_random_4d<ov::float16>(batch_num, input_f, input_y, input_x, -1, 1);
+    auto input_data_bfyx = flatten_4d(format::bfyx, input_data);
+    auto input_mem = engine.allocate_memory({ data_types::f16, format::bfyx, input_size });
+    auto in_layout = layout { ov::PartialShape::dynamic(4), data_types::f16, format::bfyx };;
+    set_values(input_mem, input_data_bfyx);
+
+    auto weights_size = tensor(output_f, input_f, filter_y, filter_x, 1);
+    auto weights_data = rg.generate_random_4d<ov::float16>(output_f, input_f, filter_y, filter_x, -1, 1);
+    auto weights_data_bfyx = flatten_4d(format::bfyx, weights_data);
+    auto weights_mem = engine.allocate_memory({ data_types::f16, format::bfyx, weights_size });
+    set_values(weights_mem, weights_data_bfyx);
+
+    // Will be used to store reference values calculated in branches depending on bias
+    auto expected_result = VVVVF<ov::float16>(batch_num, VVVF<ov::float16>(output_f));
+    topology topology;
+
+    // Calculate reference values
+    for (auto bi = 0; bi < batch_num; ++bi) {
+        for (auto ofi = 0; ofi < output_f; ++ofi) {
+            expected_result[bi][ofi] = reference_convolve(input_data[bi],                      // input
+                                                            weights_data[ofi],                 // weights
+                                                            stride, stride,                    // strides
+                                                            0,                                 // bias
+                                                            1, 1,                              // dilation
+                                                            0, 0,  // input padding
+                                                            0, 0);   // output_padding
+        }
+    }
```

Comment:
what about implementing it in `convolution_fusion_test.cpp`? I guess it would be simpler there because we have code to compare against `unfused graph`.

Reply:
Frankly, I don't think it's a good idea. Because this test has nothing to do with fusion, and probably it would be better to simplify the test code through refactoring in convolution_gpu_test later.